pkg/kubernetes/scripts/virtual-machines/components/createVmButton.jsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+
+import React from "react";
+import DialogPattern from 'cockpit-components-dialog.jsx';
+import CreateVmDialog from './createVmDialog.jsx'
+import { vmCreate } from '../kube-middleware.jsx';
+
+import { mouseClick } from '../utils.jsx'
+
+const _ = cockpit.gettext;
+
+
+class CreateVmButton extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.createVmDialog = this.createVmDialog.bind(this);
+    }
+
+    createVmDialog() {
+        let getResult;
+        let validate;
+
+        const dialogBody = (
+            <CreateVmDialog setValidateCallback={(callback) => validate = callback}
```

Comment:
It would be nice to use "reference to class component" ([doc][1]) here.

[1]: https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-class-component